### PR TITLE
Fix bug 582: Re-enable context support

### PR DESCRIPTION
**Pull request by eladiw**

### CODE REVIEW

File: presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import os
+import logging
+import spacy
+from analyzer.context_simplifier.context_simplifier import ContextSimplifier
+
+
+class NLPContextSimplifier(ContextSimplifier):
+    """ NLPContextSimplifier is responsible for preprocessing the context text
+        before the different recognizers use the context to get better
+        confidence on the score. It uses a NLP model to Rremove punctuation,
```
Comment (omri374) [inline presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py]: typo

File: presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        stop words and take lemma form and remove duplicates
+    """
```
Comment (omri374) [inline presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py]: some typos
```suggestion
    """ NLPContextSimplifier is responsible for preprocessing the context text
        (keywords and phrases provided as likely to increase patterns strength) 
        before the different recognizers use the context to get better
        confidence on the score. It uses an NLP model to lemmatize each word,
        remove stop words, punctuation and duplicates.
    """
```

File: presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import os
+import logging
+import spacy
+from analyzer.context_simplifier.context_simplifier import ContextSimplifier
```
Comment (omri374) [inline presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py]: better to use __init__ to have more clear imports 
```suggestion
from analyzer.context_simplifier import ContextSimplifier
```

File: presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    """
+
+    def __init__(self):
+        super().__init__()
+        loglevel = os.environ.get("LOG_LEVEL", "INFO")
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(loglevel)
+
+        self.logger.info("Loading NLP model (simplifier)...")
+        self.nlp = spacy.load("en_core_web_lg", disable=['parser', 'tagger'])
```
Comment (omri374) [inline presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py]: Should be injected

Comment (omri374) [inline presidio-analyzer/analyzer/context_simplifier/nlp_context_simplifier.py]: Add 'ner' to disable

File: presidio-analyzer/analyzer/pattern_recognizer.py [hunk 5]
```diff
@@ -12,8 +14,14 @@
...
 class PatternRecognizer(LocalRecognizer):
 
-    def __init__(self, supported_entity, name=None, supported_language='en',
-                 patterns=None,
+    CONTEXT_SIMILARITY_THRESHOLD = 0.65
+    CONTEXT_SIMILARITY_FACTOR = 0.35
+    MIN_SCORE_WITH_CONTEXT_SIMILARITY = 0.6
+    CONTEXT_PREFIX_COUNT = 5
+    CONTEXT_SUFFIX_COUNT = 0
+
```
Comment (omri374) [inline presidio-analyzer/analyzer/pattern_recognizer.py]: These should be in the entity_recognizer as there could be cases where we use context for non-pattern recognizers

File: presidio-analyzer/analyzer/pattern_recognizer.py [hunk 6]
```diff
@@ -73,6 +81,69 @@ def __black_list_to_regex(black_list):
...
+        # if no simplifier is defined, just tokenize the text for best effort
+        if simplifier is None:
+            return context.split()
+
+        # if a simplifier is defined, use it.
+        return simplifier.simplify(context)
+
+    @staticmethod
+    def __extract_context(text, start, end):
+        """Extract context for a specified match
```
Comment (omri374) [inline presidio-analyzer/analyzer/pattern_recognizer.py]: We're doing this multiple times even though the text doesn't necessarily change between recognizers. I think it should be a post-process but let's discuss this offline

File: presidio-analyzer/requirements.txt [hunk 7]
```diff
@@ -1,6 +1,6 @@
 cython
 https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-2.0.0/en_core_web_lg-2.0.0.tar.gz
-https://github.com/torosent/pyre2/archive/release/0.2.23.zip
+regex
```
Comment (omri374) [inline presidio-analyzer/requirements.txt]: This would completely remove pyre2 which is faster than regex

**Comment by eladiw** [comment]
Closing PR . will reopen when work is done

**Pull request closed without merging**